snippets/curl/nb_images.sh

```diff
@@ -0,0 +1,79 @@
+!/bin/sh
+
+#####################n#####################################################
+
+# This script demostrates the usage of netbackup REST API for listing 
+# the backup images
+
+# This script requires jq command-line JSON parser 
+# if your system does not have jq installed, this will not work
+
+###########################################################################
+
+port=1556
+master_server=""
+username=""
+password=""
+
+showHelp()
+{
+	echo ""
+	echo "Invalid command parameters"
+	echo "Usage:"
+	echo "./nb_images.sh -master <master_server> -username <username> -password <password>"
+	echo ""
+	exit 1
+}
+
+parseArguments()
+{
+	if [ $# -lt 6 ]; then
+		echo "Invalid number of arguments"
+		showHelp
+	fi
+	
+	while [ "$1" != "" ]; do
+		case $1 in
+			-master)
+				master_server=$2
+				;;
+			-username)
+				username=$2
+				;;
+			-password)
+				password=$2
+				;;
+			*)
+				showHelp
+				;;
+		esac
+		shift 2
+	done
+	
+	if [ -z "$master_server" ] || [ -z "$username" ] || [ -z "$password" ]; then
+		showhelp
+	fi
+}
+
+#main
+parseArguments "$@"
+
+basepath="https://$master_server:$port/netbackup"
+content_header='content-type:application/json'
+
+##############login#############
+
+uri="$basepath/login"
+data="{\"userName\":\"$username\",\"password\":\"$password\"}"
+
+jwt=$(curl -k -X POST $uri -H $content_header -d $data | jq '.token' | sed "s/\"//g")
+
+##############jobs##############
+
+auth_header="authorization:$jwt"
+uri="$basepath/catalog/images"
+
+curl -k -X GET $uri  -H $content_header -H $auth_header|jq '[.data[]|{IMAGE_ID: .id, POLICY: .attributes.policyName, CLIENT: .attributes.clientName, BACKUP_TIME: .attributes.backupTime}]'
```

Comment:
Minor: Do you want to also include an example of using filter?

Reply:
added filters